frontend/src/features/tasks/state/TaskStateManager.ts

```
@@ -24,13 +24,16 @@ class TaskStateManagerImpl {
   private deps: TaskStateMachineDeps | null = null
   private globalListeners: Set<(taskId: number, state: TaskStateData) => void> = new Set()
+  private initListeners: Set<() => void> = new Set()
 
   /**
    * Initialize the manager with dependencies from SocketContext
    * This MUST be called before using any other methods
    */
   initialize(deps: TaskStateMachineDeps): void {
     this.deps = deps
+    // Notify init listeners so useSyncExternalStore picks up the change
+    this.initListeners.forEach(listener => listener())
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)